### PR TITLE
Avoid memory issue if one of the write plugins is slow/unavailable (e.g. graphite) by limiting the queue size

### PR DESCRIPTION
**Pull request by puckel**

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -5,6 +5,9 @@
 default["collectd"]["checksum"]           = "853680936893df00bfc2be58f61ab9181fecb1cf45fc5cddcb7d25da98855f65"
 default["collectd"]["interval"]           = 10
 default["collectd"]["read_threads"]       = 5
+default['collectd']['WriteQueueLimitHigh'] = 2_000_000
```
Comment (hectcastro) [inline attributes/default.rb]: Can you please make these changes consistent with all of the other attributes?

- Single quotes to double quotes
- Snake case names
- `'false'` as boolean

In addition, I think the cookbook should use the same default queue limits that `collectd` uses:

```
WriteQueueLimitHigh 1000000
WriteQueueLimitLow   800000

```

File: recipes/default.rb [hunk 2]
```diff
@@ -104,6 +104,9 @@
     :dir          => node["collectd"]["dir"],
     :interval     => node["collectd"]["interval"],
     :read_threads => node["collectd"]["read_threads"],
+    WriteQueueLimitHigh: node['collectd']['WriteQueueLimitHigh'],
```
Comment (hectcastro) [inline recipes/default.rb]: Same here. Can you please make these consistent with the other attributes?

- Single quotes to double quotes
- Snake case names
- Old style symbols

**Comment by hectcastro** [comment]
Thanks for the contribution, @puckel. I left a few comments, but otherwise these additions look good.

**Comment by puckel** [comment]
I've did the changes you mention. Previously, I did my change to respect rubocop/foodcritic from the last version of ChefDK

**Comment by hectcastro** [comment]
Ah, I have not done a good job of keeping up with all of the various code analysis tooling used by the Chef community.

I appreciate that you made these changes, and I opened https://github.com/hectcastro/chef-collectd/issues/55 to help get everything using the new syntax and tooling.

**Pull request merged**